packages/apollo-shared/src/ChangeManager/ChangeManager.ts

```diff
@@ -86,9 +99,81 @@ export class ChangeManager {
     }
   }
 
+  async changeThroughSocket(change: Change) {
+    let ch
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const tmpObject: any = {
+      ...change,
+    }
+    const { featureId, changedIds, assembly } = tmpObject
+
+    console.log(`CHANGE MANAGER: OBJECT: ${JSON.stringify(tmpObject)}`)
+    switch (change.typeName) {
+      case 'AddAssemblyFromFileChange':
+        ch = new AddAssemblyFromFileChange({
+          typeName: 'AddAssemblyFromFileChange',
+          assembly: tmpObject.assembly,
+          assemblyName: tmpObject.assemblyName,
+          changes: tmpObject,
+        })
+        break
+      case 'AddFeatureChange':
+        const { addedFeature } = tmpObject
+        ch = new AddFeatureChange({
+          typeName: 'AddFeatureChange',
+          changedIds,
+          addedFeature,
+          assembly,
+          parentFeatureId: tmpObject.parentFeatureId,
+        })
+        break
+      case 'DeleteAssemblyChange':
+        ch = new DeleteAssemblyChange({
+          typeName: 'DeleteAssemblyChange',
+          assembly,
+        })
+        break
+      case 'DeleteFeatureChange':
+        const { parentFeatureId, deletedFeature } = tmpObject
+        ch = new DeleteFeatureChange({
+          typeName: 'DeleteFeatureChange',
+          changedIds,
+          deletedFeature,
+          parentFeatureId,
+          assembly,
+        })
+        break
+      case 'LocationEndChange':
+        const { oldEnd, newEnd } = tmpObject
+        ch = new LocationEndChange({
+          typeName: 'LocationEndChange',
+          changedIds,
+          featureId,
+          oldEnd,
+          newEnd,
+          assembly,
+        })
+        break
+      case 'LocationStartChange':
+        const { oldStart, newStart } = tmpObject
+        ch = new LocationStartChange({
+          typeName: 'LocationStartChange',
+          changedIds,
+          featureId,
+          oldStart,
+          newStart,
+          assembly,
+        })
+        break
+    }
+
+    // submit to client data store
+    await ch?.apply(this.dataStore)
+  }
+
   async revert(change: Change, submitToBackend = true) {
-    const inverseChange = change.getInverse()
-    return this.submit(inverseChange, { submitToBackend, addToRecents: false })
+    change.getInverse()
```

Comment:
What is this change to `revert` for?

Reply:
I don't recognise this change either. Restored back.

Reply:
Fixed